docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,687 @@
+  "deactivated": false
+}
+
+```
+
+
+**Verification methods and relationships**
+
+A new verification method MAY be added using the gRPC message
+
+```
+MsgAddVerification(did string, accountId string, relationships []string, signerAccount string)
+```
+
+The parameter `did` identifies the did document, the `accountId` is the account to be added to the verification method, the `relationships` are the list of relationships that the `accountId` shall be registered into.  The `signerAccount` is the account address signing the transaction.
+
+The list of relationships MUST contains only valid [relationships names](#DID_document)
+
+Example:
+
+```javascript
+/* gRPC message */
+MsgAddVerification(
+    "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb", 
+    "cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2",
+    ["authentication", "keyAgreement"],
+    "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj" // <-- the signer has authorization relationship
+)
+
+/* DID document */
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb",
+    "controller": [
+        "did:key:cosmos195rlq2hjnn2tmagskys4xtsnsey6gjljg8zxrn" 
+    ],
+    "verificationMethods" : [
+        {
+            "id": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        },
+        {   // <--  a new verification method is added 
+            "id": "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2#cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2",
+            "blockchainAccountID": "cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+        "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2#cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2"  //<-- new verification method added to authentication relationship
+    ],
+    "keyAgreement": {
+         "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2#cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2" //<-- new verification method added to keyAgreement relationship
+    }
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-05-23T06:35:22Z",  // <--  update field modified
+  "versionId": "262495b1159c1cd7faf0da56deb6521bb2980d04435818906e417aae47604027", /* hex(blake2b(tx)) */ // <-- new hash computed
+  "deactivated": false
+}
+
+```
+
+The relationships of a verification method MAY be set using the gRPC message:
```

Comment:
again, not sure why we are using MAY capital, we want to use can here

I prefer imperative language as shown here, especially for instructions

Reply:
```suggestion
Set the relationships of a verification method using the gRPC message:
```